File: Data Structures & Algorithms/jump-game/submission-1.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        goal = n - 1
        curr = n - 2
        while curr >= 0:
            if curr + nums[curr] >= goal:
                goal = curr
            curr = curr - 1

        if goal == 0:
            return True
        return False


            



            

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_map<int,int> m;
        int res = 0;

        for(auto& n : nums){
            if(!m[n]){
                m[n] = m[n - 1] + m[n + 1] + 1;
                m[n - m[n - 1]] = m[n];
                m[n + m[n + 1]] = m[n];
                res = max(res, m[n]);
            }
        }
        
        return res;
    }
};




